access/types/forms.py

```diff
@@ -233,28 +234,83 @@ def add_field(self, i, config, field_class, widget_class,
         if not choices is None:
             args['choices'] = choices
 
+        name = self.field_name(i, config)
```

Comment:
Could this new long chunk of code be moved to a new method perhaps? Just to make the code more readable and organized.